### PR TITLE
Add meta registration for foreach_maximum_.List

### PR DESCRIPTION
**Pull request by janeyx99**
Will fix issues compiling for when amsgrad is True for Adam(W), see related failures in https://github.com/pytorch/benchmark/actions/runs/5628705163/job/15252867793

Also did some refactoring where common registrations could be deduplicated.

Test plan:
python test/inductor/test_compiled_optimizers.py -k test_adam

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105864

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -2590,43 +2590,31 @@ def _check_foreach_binop_tensor_lists(self, other):
...
-
-@register_meta([aten._foreach_maximum.List])
-def meta__foreach_maximum__list(self, other, alpha=1):
+@register_meta(
+    [
+        aten._foreach_add.List,
+        aten._foreach_sub.List,
+        aten._foreach_mul.List,
+        aten._foreach_div.List,
+        aten._foreach_maximum.List,
```
Comment (crcrpar) [inline torch/_meta_registrations.py:2599]: admitting I'm not familiar with meta registration at all but wouldn't `_foreach_minimum` be worth adding as well?

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -2590,43 +2590,31 @@ def _check_foreach_binop_tensor_lists(self, other):
...
 
 @register_meta(
     [
-        aten._foreach_div.List,
-        aten._foreach_mul.List,
+        aten._foreach_add_.List,
+        aten._foreach_sub_.List,
+        aten._foreach_mul_.List,
+        aten._foreach_div_.List,
+        aten._foreach_maximum_.List,
```
Comment (crcrpar) [inline torch/_meta_registrations.py:2613]: ditto

File: torch/_meta_registrations.py [hunk 3]
```diff
@@ -2590,43 +2590,31 @@ def _check_foreach_binop_tensor_lists(self, other):
...
     ]
 )
-def meta__foreach_binop_list(self, other):
+def meta__foreach_binop__list(self, other):
     _check_foreach_binop_tensor_lists(self, other)
-    return [torch.empty_like(s) for s in self]
```
Comment (albanD) [inline torch/_meta_registrations.py:2629]: Is no return ok here?
I know it matches the actual impl but not sure compile will be happy with that haha

**Review by albanD — APPROVED** [review]
SGTM if CI is happy and minimum added.
Also lerp, etc?

File: torch/_meta_registrations.py (hunk 3 above)
Comment (janeyx99) [reply torch/_meta_registrations.py:2629]: No return is what's expected from inplace foreach ops so this actually can't return stuff :/

File: torch/_meta_registrations.py (hunk 2 above)
Comment (janeyx99) [reply torch/_meta_registrations.py:2613]: Oki! Thanks :D

**Comment by janeyx99** [comment]
lerp is weird and has its own registrations. I'll also add some tests.

**Review by mlazos — APPROVED** [review]

File: torch/_meta_registrations.py (hunk 3 above)
Comment (albanD) [reply torch/_meta_registrations.py:2629]: Ok ok

**Pull request closed without merging**